tests/integration/relationships/nested-relationship-test.js

```diff
@@ -35,6 +37,25 @@ module('integration/relationships/nested_relationships_test - Nested relationshi
       adapter: DS.JSONAPIAdapter
     });
 
+    env.registry.register('adapter:middle-ager', DS.JSONAPIAdapter.extend({
+      findHasMany (store, snapshot, url, relationship) {
+        return new Promise((resolve) => {
+          run.later(() => {
+            if (url === '/middle-agers/1/kids') {
+              resolve({
+                data: [{
+                  id: 1,
+                  type: 'kid'
+                }]
+              });
+            } else {
+              resolve({ data: [] });
+            }
+          }, 10);
```

Comment:
Same story here. Returning `Promise.resolve(...)` in each branch of this conditional should allow us to remove the run.later.